Model/Checkout/PlaceOrder.php

```diff
@@ -412,21 +412,28 @@ public function verifyRefId(string $refId, RefIdDataInterface $refIdData): bool
     }
 
     /**
+     * Apply discount rule based on card bin prefix.
+     *
      * @param CartInterface $quote
      * @param string $card_bin
      * @return void
      */
     private function applyBinDiscountRule(CartInterface $quote, string $card_bin): void
     {
         $selectedDiscount = $this->config->getDiscountRule();
-        $cards_bin_codes = $this->config->getCardsBinCodes();
         if (!$selectedDiscount) {
             return;
         }
-        if (!in_array($card_bin, $cards_bin_codes)) {
-            return;
+        $cards_bin_codes = $this->config->getCardsBinCodes();
+        foreach ($cards_bin_codes as $bin) {
+            if (!ctype_digit($bin)) {
+                continue; // Skip if $bin is not purely numeric
+            }
+            if ( str_starts_with($card_bin, $bin) ) {
```

Comment:
For backwards compatability with PHP versions older than 8
```suggestion
            if (strpos($cardBin, $bin) === 0 ) {
```